### PR TITLE
[J2C+] QoS/PFC test fixes for SONIC chassis

### PR DESCRIPTION
**Pull request by vmittal-msft**
### Description of PR

Summary:
Fixes # (issue)

### Type of change
QoS test fixes for SONIC chassis

- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [x] Test case(new/improvement)

### Back port request
- [ ] 201911
- [ ] 202012
- [x] 202205

#### What is the motivation for this PR?
QoS/PFC test case fixes for SONIC chassis

#### How did you do it?
Updated test params as well as test infra improvements

#### How did you verify/test it?
On J2C+ based DUT. Also, verified these changes are not impacting existing T0 toplogies.

#### Supported testbed topology if it's a new test case?
T2

### Documentation

Related to https://github.com/sonic-net/sonic-mgmt/pull/6946

### CODE REVIEW

File: tests/common/devices/sonic.py [hunk 1]
```diff
@@ -1521,7 +1521,9 @@ def show_and_parse(self, show_cmd, **kwargs):
     @cached(name='mg_facts')
     def get_extended_minigraph_facts(self, tbinfo, namespace=DEFAULT_NAMESPACE):
         mg_facts = self.minigraph_facts(host=self.hostname, namespace=namespace)['ansible_facts']
-        mg_facts['minigraph_ptf_indices'] = mg_facts['minigraph_port_indices'].copy()
+        mg_facts['minigraph_ptf_indices'] = {}
+        if tbinfo["topo"]["name"] != 't2_2lc_min_ports-masic':
+            mg_facts['minigraph_ptf_indices'] = mg_facts['minigraph_port_indices'].copy()
```
Comment (sanmalho-git) [inline tests/common/devices/sonic.py:1526]: Why is the above copy, not valid for 't2_2lc_min_ports-masic'?

Comment (vmittal-msft) [reply tests/common/devices/sonic.py:1526]: For min topology case, when we copy port info to ptf info and update correct ptf indices later, we get into duplicate values for different keys. This gives us issue when we try to get port_name from value. Here is the example during the test -

u'Ethernet0': 0
u'Ethernet8': 1
u'Ethernet16': 6
u'Ethernet24': 7
u'Ethernet32': 4
u'Ethernet40': 5
u'Ethernet48': 8
u'Ethernet64': 8
u'Ethernet72': 9,
u'Ethernet80': 10
 u'Ethernet96': 12
u'Ethernet192': 9
u'Ethernet200': 10

Comment (sanmalho-git) [reply tests/common/devices/sonic.py:1526]: But, if we have using the correct topo file and ports, the duplicate entries will never be accessed.

**Review by sanmalho-git — DISMISSED** [review]

**Review by sanmalho-git — APPROVED** [review]

**Review by tjchadaga — APPROVED** [review]

**Comment by StormLiangMS** [comment]
@vmittal-msft this commit break the T1 test_fib case, below is the err msg
 UnboundLocalError: local variable 'dut_port' referenced before assignment.

**Pull request merged**